packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -196,15 +191,13 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
         }
     } */
 
-    async getSigner<Type extends SignerType>(type: Type): Promise<SignerMap[Type]> {
+    getSigner<Type extends SignerType>(type: Type): SignerMap[Type] {
         switch (type) {
             case "viem": {
-                const walletClient = createWalletClient({
-                    account: this.account,
-                    chain: getViemNetwork(this.chain as EVMBlockchainIncludingTestnet),
-                    transport: http(getBundlerRPC(this.chain)),
-                }).extend(publicActions) as any;
-                return walletClient as SignerMap[Type];
+                return {
+                    public: this.publicClient,
+                    wallet: this.kernelClient,
```

Comment:
<img width="689" alt="image" src="https://github.com/Crossmint/crossmint-sdk/assets/103966567/0ae4729a-35ed-4aca-864c-801dd0ce374e">
```suggestion
                    publicClient: this.publicClient,
                    walletClient: this.kernelClient,
```